luda-adventure/src/ecs/component.rs

```diff
@@ -0,0 +1,68 @@
+use super::Entity;
+use std::fmt::Debug;
+
+pub trait Component: Debug {}
+
+#[derive(Debug)]
+pub struct ComponentContainer<T> {
+    pub component: T,
+}
+impl<T> ComponentContainer<T>
+where
+    T: Component,
+{
+    pub fn new(component: T) -> Self {
+        Self { component }
+    }
+    pub fn as_ref(&self) -> &T {
+        &self.component
+    }
+    pub fn as_ref_mut(&mut self) -> &mut T {
+        &mut self.component
+    }
+}
+
+pub trait ContainedComponent: Debug + 'static {
+    fn as_any(&self) -> &dyn std::any::Any;
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
+}
```

Comment:
Why did you need `ComponentContainer` and `ContainedComponent`, not only just `Component`?

Reply:
For downcasting. If we store them as `Vec<Box<dyn component>>`, It's typeid will be `Box<dyn component>`'s, not specific component's like `Box<Mover>`.